Detectors/MUON/MCH/Align/src/AlignmentSpec.cxx

```diff
@@ -0,0 +1,1096 @@
+    vector<o2::detectors::AlignParam> ParamsRef;
+    o2::detectors::AlignParam param_Ref;
+
+    for (int hc = 0; hc < 20; hc++) {
+
+      ParamsRef.emplace_back(ParamsTrack.at(fgSNDetElemHalfCh[hc]));
+
+      for (int de = 0; de < fgNDetElemHalfCh[hc]; de++) {
+
+        int iDEN = fgDetElemHalfCh[hc][de];
+        o2::detectors::AlignParam param_Track = ParamsTrack.at(fgSNDetElemHalfCh[hc] + 1 + de);
+
+        LOG(debug) << Form("%s%s", "Processing DET Elem: ", (param_Track.getSymName()).c_str());
+
+        TGeoHMatrix delta_track;
+        TGeoRotation r("Rotation/Track", param_Track.getPsi() / pi() * 180.0, param_Track.getTheta() / pi() * 180.0, param_Track.getPhi() / pi() * 180.0);
+        delta_track.SetRotation(r.GetRotationMatrix());
+        delta_track.SetDx(param_Track.getX());
+        delta_track.SetDy(param_Track.getY());
+        delta_track.SetDz(param_Track.getZ());
+
+        TGeoHMatrix transRef = transformIdeal[iDEN];
+        TGeoHMatrix transTrack = doReAlign ? transformNew[iDEN] : transformRef[iDEN];
+        TGeoHMatrix transRefTrack = transTrack * transRef.Inverse();
+        TGeoHMatrix delta_ref = delta_track * transRefTrack;
+
+        param_Ref.setSymName((param_Track.getSymName()).c_str());
+        param_Ref.setGlobalParams(delta_ref);
+        param_Ref.applyToGeometry();
+        ParamsRef.emplace_back(param_Ref);
+      }
+    }
+
+    TFile* fOut = TFile::Open("AlignParam@ideal.root", "RECREATE");
+    fOut->WriteObjectAny(&ParamsRef, "std::vector<o2::detectors::AlignParam>", "alignment");
+    fOut->Close();
+  }
+
+  //_________________________________________________________________________________________________
+  Track MCHFormatConvert(TrackMCH& mchTrack, vector<Cluster>& mchClusters, bool doReAlign)
+  {
+
+    Track convertedTrack = Track();
+
+    // Get clusters for current track
+    int id_cluster_first = mchTrack.getFirstClusterIdx();
+    int id_cluster_last = mchTrack.getLastClusterIdx();
+
+    for (int id_cluster = id_cluster_first;
+         id_cluster < id_cluster_last + 1; ++id_cluster) {
+
+      Cluster* cluster = &(mchClusters.at(id_cluster));
+      const int DEId_cluster = cluster->getDEId();
+      const int CId_cluster = cluster->getChamberId();
+      const int ind_cluster = cluster->getClusterIndex();
+
+      // Transformations to new geometry from reference geometry
+      if (doReAlign) {
+
+        math_utils::Point3D<double> local;
+        math_utils::Point3D<double> master;
+
+        master.SetXYZ(cluster->getX(), cluster->getY(), cluster->getZ());
+
+        transformRef[cluster->getDEId()].MasterToLocal(master, local);
+        transformNew[cluster->getDEId()].LocalToMaster(local, master);
+
+        cluster->x = master.x();
+        cluster->y = master.y();
+        cluster->z = master.z();
+      }
+      convertedTrack.createParamAtCluster(*cluster);
+    }
+
+    return Track(convertedTrack);
```

Comment:
There is no need for this extra copy --> `return convertedTrack;`

Reply:
I tried to do this, but compilation fails with error:
`
/Users/cz263673/alice/sw/SOURCES/O2/mchAlign_Chi/0/Detectors/MUON/MCH/Align/src/AlignmentSpec.cxx:575:12: error: call to deleted constructor of 'Track'
    return convertedTrack;
           ^~~~~~~~~~~~~~
/Users/cz263673/alice/sw/SOURCES/O2/mchAlign_Chi/0/Detectors/MUON/MCH/Tracking/include/MCHTracking/Track.h:40:3: note: 'Track' has been explicitly marked deleted here
  Track(Track&&) = delete;
  ^
`

Reply:
ok, I see.